packages/code-infra/src/cli/cmdPublishCanary.mjs

```diff
@@ -161,17 +393,27 @@ async function publishCanaryVersions(
   } finally {
     // Always restore original package.json files in parallel
     console.log('\n🔄 Restoring original package.json files...');
-    const restorePromises = allPackages.map(async (pkg) => {
-      const originalPackageJson = originalPackageJsons.get(pkg.name);
-      await writePackageJson(pkg.path, originalPackageJson);
-    });
+    const restorePromises = updateResults.map(
+      async ({ pkg, originalPackageJson, pkgJsonDirectory }) => {
+        // no need to restore package.json files in build directories
```

Comment:
But it doesn't matter if we restore it anyway, right? The overhead wouldn't really be noticable, and it can save us from having to leak `pkgJsonDirectory`. To reduce complexity.